### PR TITLE
Frontend testing

### PR DESCRIPTION
**Pull request by badnames**
## Basics

- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.
- [x] The PR is rebased with current master.

## Checklist

- [x] I added unit tests for my code
- [x] I fully described what my PR does in the documentation
      (not in the PR description)
- [x] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [x] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

File: frontend/src/components/Button/__snapshots__/SimpleButton.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`renders correctly for primary base variant 1`] = `
+<button
+  className="inline-flex h-10 w-full items-center justify-center rounded-lg text-white px-4 py-2.5 text-center text-sm font-medium focus:outline-none focus:ring-4 focus:ring-blue-300 border bg-primary-500 dark:bg-primary-300 border-primary-500 hover:bg-primary-600 dark:hover:bg-primary-200 active:bg-primary-900 text-primary-50 dark:text-primary-700"
```
Comment (markus2330) [inline frontend/src/components/Button/__snapshots__/SimpleButton.test.tsx.snap:5]: Do we really want so precise checks? This looks like a maintenance effort on every cosmetic change? Or is it so easy to redo the snapshots or useful in some way? Can you please write a README how to do it?

If you learned something during testing from the code, please add code comments.

Comment (badnames) [reply frontend/src/components/Button/__snapshots__/SimpleButton.test.tsx.snap:5]: These are automatically generated. Apparently you can easily regenerate them if a component was changed.

**Comment by badnames** [comment]
@markus2330 To prevent this PR from growing way too large *again*, would it be feasible to move the creation of testcases to a separate PR?

I will finish the testcases I have begun to implement in the components folder for completeness sake and to provide some examples.

**Comment by markus2330** [comment]
> To prevent this PR from growing way too large again, would it be feasible to move the creation of testcases to a separate PR?

Of course, you can always split up PRs as needed.

> Documentation on how to create and run testcases will of course also be included in this PR.

:+1:

File: frontend/README.md [hunk 2]
```diff
@@ -54,6 +54,21 @@ npm run storybook
...
+```shell
+npm run test
+```
+
+### Testing individual files
+
+```shell
+npm run test -- path/to/component.test.tsx
+```
+
```
Comment (Bushuo) [inline frontend/README.md:71]: We would also explain how to run the tests in watch mode which I think is an immensly useful feature.

I can't suggest a code block in with the suggestions here :/

### Testing in watch mode

To execute those tests for which changes to the code are detected during development.
```shell
npm run test -- --watch
```

**Review by Bushuo — APPROVED** [review]
I think this looks good. 
I am also curious of how well snapshot testing works. I want to give it a try.
👍

**Review by absurd-turtle — APPROVED** [review]
The snapshot test look promising for testing layouts! I haven't used them before. Maybe you can give a quick introduction in the next frontend meeting.

File: frontend/README.md (hunk 2 above)
Comment (badnames) [reply frontend/README.md:71]: This is a great suggestion. I will add it asap

**Pull request merged**